docs/word/word-add-ins-core-concepts.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Fundamental programming concepts with the Word JavaScript API
+description: Use the Word JavaScript API to build add-ins for Word.
+ms.date: 07/05/2019
+localization_priority: Priority
+---
+
+# Fundamental programming concepts with the Word JavaScript API
+
+This article describes concepts that are fundamental to using the [Word JavaScript API](../reference/overview/word-add-ins-reference-overview.md) to build add-ins for Word 2016 or later.
+
+## Asynchronous nature of Word APIs
+
+The Word JavaScript API is loaded by Office.js. The Word JavaScript API changes the way that you can interact with objects like documents and paragraphs. Rather than providing individual asynchronous APIs for retrieving and updating each of these objects, the Word JavaScript API provides "proxy" JavaScript objects that correspond to the real objects running in Word. You can interact with these proxy objects by synchronously reading and writing their properties and calling synchronous methods to perform operations on them. These interactions with proxy objects aren't immediately realized in the running script. The **context.sync** method synchronizes the state between your running JavaScript and the real objects in Office by executing queued instructions and retrieving properties of loaded Word objects for use in your script.
```

Comment:
Is it worth mentioning `load` here? Maybe this section needs to go after the "Referencing Office.js" and "Running Word add-ins" sections (right before the "Synchronizing..." section).

Reply:
I've re-sequenced the sections -- good suggestion.

FYI - although GitHub shows this entire article as "new" content -- it was all copied verbatim from the previous version of the Word API ref intro page. I made a few changes (to fix easily-fixed problems), but there are still many other improvements than can be made to this article. This work is beyond the scope of this PR, but I'm tracking it with a separate work item in VSO.